RecoLocalTracker/SiStripRecHitConverter/src/SiStripTemplate.cc

```diff
@@ -552,7 +552,7 @@ bool SiStripTemplate::pushfile(const SiPixelTemplateDBObject& dbobject, std::vec
 				qavg_avg = 0.f;
 				for (j=0; j<9; ++j) {
 					
-					for (l=0; l<TSXSIZE; ++k) {db >> theCurrentTemp.entx[k][i].xtemp[j][l]; qavg_avg += theCurrentTemp.entx[k][i].xtemp[j][l];} 
+					for (l=0; l<TSXSIZE; ++l) {db >> theCurrentTemp.entx[k][i].xtemp[j][l]; qavg_avg += theCurrentTemp.entx[k][i].xtemp[j][l];} 
```

Comment:
This fix, while due, seems to possible affect the normal workflow of the code: when k was (erroneously) incremented, the wrong k entered the following checks inside the outer loop
@OlivierBondu : whenever there are no issues with what stored in the the db probably the previous bug didn't show up, Do you expect situations in which there is any effect of this fix? Where should we look at to possibly see one?

Reply:
there is NOT such a thing as a StripTemplate in the database....

Reply:
Do you mean that this part of the code was never hit, and therefore while the fix is due we cannot see any effect out of it?

Reply:
Naively thought the effect of the bug should be an infinite loop (likely leading to a segfault at some point).

Reply:
Indeed... That's correct @makortel: this demonstrates that this code was never run!

Reply:
I had a quick look, and it seems there are two reasons why this did not give problems:
- the template CPE is not used for the strips (StripCPEfromTrackAngle is the default), and the template cfi is only imported in some outdated test configs
- there is no code in CMSSW that calls this method (the template CPE uses the one defined above, `bool SiStripTemplate::pushfile(int filenum, std::vector< SiStripTemplateStore > & theStripTemp_)`) - I verified by removing it and recompiling (after `git cms-checkdeps -a`), which showed no problems.

According to git this has not been touched since the migration from CVS. I think it has been copied/adapted from the corresponding code for the pixel detector [here](https://github.com/cms-sw/cmssw/blob/225ca5009bec119907444f0f79c5036f9b0aad67/RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplate.cc#L551) at some point.
Question (also to @mmusich @echabert @boudoul ): since this is not used (and apparently never has been), and it can be revived from git history or the pixel version in case it is needed some day, should we just remove this method?

Reply:
Hello, @pieterdavid, @perrotta,
as you correctly point out Strip template reconstruction was never developed (same people working on the pixel one were involved here) up to a point where it could be used in production and certainly we don't even have `CondFormat` to be able to store in DB. While we need to support the `fromTrackAngle` and, up to a certain degree the `geometric` CPE algorithm, I personally see no reason to keep in repository the template code at all if all it does is to create maintenance overhead.

Reply:
I second @mmusich 's comment

Reply:
@pieterdavid  @mmusich @echabert @boudoul: will any of you take care of preparing a PR which removes from the release whatever is not used (in this package)? Thanks!

Reply:
Thanks @mmusich and @boudoul , I was only talking about this specific method, but I'm not against removing it completely either... @perrotta : submitted as https://github.com/cms-sw/cmssw/pull/22546